frontend/packages/kubevirt-plugin/integration-tests/tests/vm.environment.scenario.ts

```diff
@@ -0,0 +1,126 @@
+import { getSecret, getConfigMap, getServiceAccount } from './utils/mocks';
+import { createResource, deleteResource } from '@console/shared/src/test-utils/utils';
+import { browser, ExpectedConditions as until, Key } from 'protractor';
+import { createExampleVMViaYAML } from './utils/utils';
+import { testName, appHost } from '@console/internal-integration-tests/protractor.conf';
+import { isLoaded } from '@console/internal-integration-tests/views/crud.view';
+import * as vmEnv from '../views/vm.environment.view';
+import { addVariableFrom } from '@console/internal-integration-tests/views/environment.view';
+import {
+  PAGE_LOAD_TIMEOUT_SECS,
+  KUBEVIRT_SCRIPTS_PATH,
+  VM_BOOTUP_TIMEOUT_SECS,
+} from './utils/consts';
+import { execSync } from 'child_process';
+
+const expecScriptPath = `${KUBEVIRT_SCRIPTS_PATH}/expect-vm-env-readable.sh`;
+const vmName = 'vm-example';
+const configmapName = 'configmap-mock';
+const secretName = 'secret-mock';
+const serviceAccountName = 'service-account-mock';
+
+describe('Test VM enviromnet tab', () => {
+  const secret = getSecret(testName, secretName);
+  const configmap = getConfigMap(testName, configmapName);
+  const serviceAccount = getServiceAccount(testName, serviceAccountName);
+
+  beforeAll(async () => {
+    createResource(secret);
+    createResource(configmap);
+    createResource(serviceAccount);
+    createExampleVMViaYAML();
+  });
+
+  afterAll(() => {
+    deleteResource(secret);
+    deleteResource(configmap);
+    deleteResource(serviceAccount);
+  });
+
+  beforeEach(async () => {
+    await browser.get(`${appHost}/k8s/ns/${testName}/virtualmachines/${vmName}/environment`);
+    await isLoaded();
+  });
+
+  it('Add configmap, secret and service account', async () => {
+    await vmEnv.addSource(configmapName);
+    await vmEnv.addSource(secretName);
+
+    // Add Service Account
+    await vmEnv.clickAddSourceButton();
+    await addVariableFrom(serviceAccountName, null, true);
+    await browser.wait(until.presenceOf(vmEnv.successAlert));
+    expect(vmEnv.successAlert.isDisplayed()).toEqual(true);
+  });
+
+  it('Verify all the sources are present at the disks tab', async () => {
+    await browser.get(`${appHost}/k8s/ns/${testName}/virtualmachines/${vmName}/disks`);
```

Comment:
With the VirtualMachine instance, this can be substituted with 

```
const disks = await vm.getAttachedDisks();

expect(disks).toContain( <configMapDisk> ); // you need to have configMapDisk in StorageResource type, or you can just search by name


```